pallets/ddc-verification/src/tests.rs

```diff
@@ -1228,3 +1275,283 @@ fn test_get_consensus_nodes_activity_diff_errors() {
+			(NodePubKey::StoragePubKey(StorageNodePubKey::new([1; 32])), node_params1),
+			(NodePubKey::StoragePubKey(StorageNodePubKey::new([2; 32])), node_params2),
+			(NodePubKey::StoragePubKey(StorageNodePubKey::new([3; 32])), node_params3),
+			(NodePubKey::StoragePubKey(StorageNodePubKey::new([4; 32])), node_params4),
+		];
+
+		let dac_account = AccountId::from([1; 32]);
+		let cluster_id = ClusterId::from([12; 20]);
+		let era = 16;
+		let merkel_root_hash: H256 = array_bytes::hex_n_into_unchecked(
+			"95803defe6ea9f41e7ec6afa497064f21bfded027d8812efacbdf984e630cbdc",
+		);
+
+		assert_noop!(
+			Pallet::<Test>::get_era_to_validate(cluster_id, dac_nodes.clone()),
+			Error::<Test>::NoValidatorExist
+		);
+
+		ValidatorSet::<Test>::put(vec![dac_account.clone()]);
+
+		assert_ok!(DdcVerification::set_current_validator(RuntimeOrigin::signed(
+			dac_account.clone()
+		),));
+
+		assert_noop!(
+			Pallet::<Test>::get_era_to_validate(cluster_id, dac_nodes.clone()),
+			Error::<Test>::EraToValidateRetrievalError
+		);
+
+		assert_ok!(DdcVerification::create_billing_reports(
+			RuntimeOrigin::signed(dac_account.clone()),
+			cluster_id,
+			era,
+			merkel_root_hash,
+			merkel_root_hash,
+		));
+
+		let result = Pallet::<Test>::get_era_to_validate(cluster_id, dac_nodes.clone());
+		assert!(result.is_ok());
+		assert_eq!(result.unwrap().unwrap(), era_activity2.id); //17
+	});
+}
+
+#[test]
+fn off_chain_worker_works() {
+	let mut ext = TestExternalities::default();
+	let (offchain, _offchain_state) = TestOffchainExt::new();
+	let (pool, pool_state) = TestTransactionPoolExt::new();
+	let cluster_id = ClusterId::from([12; 20]);
+	let era = 16;
+	let merkel_root_hash: H256 = array_bytes::hex_n_into_unchecked(
+		"95803defe6ea9f41e7ec6afa497064f21bfded027d8812efacbdf984e630cbdc",
+	);
+	let (pair, _seed) = sp_core::sr25519::Pair::from_phrase(
+		"spider sell nice animal border success square soda stem charge caution echo",
+		None,
+	)
+	.unwrap();
+	let keystore = MemoryKeystore::new();
+	keystore
+		.insert(
+			KEY_TYPE,
+			"0xb6186f80dce7190294665ab53860de2841383bb202c562bb8b81a624351fa318",
+			pair.public().as_ref(),
+		)
+		.unwrap();
+
+	ext.register_extension(OffchainWorkerExt::new(offchain.clone()));
+	ext.register_extension(OffchainDbExt::new(offchain));
+	ext.register_extension(TransactionPoolExt::new(pool));
+	ext.register_extension(KeystoreExt::new(keystore));
+
+	ext.execute_with(|| {
+		// Offchain worker should not be triggered if block number is not divided by 100
+		let block = 102;
+		System::set_block_number(block);
+
+		DdcVerification::offchain_worker(block);
+		assert_eq!(pool_state.write().transactions.pop(), None);
+
+		// // Offchain worker should be triggered if block number is  divided by 100
+		let block = 500;
+		System::set_block_number(block);
+		let validator: AccountId = AccountId::from(pair.public().0);
+		<ValidatorSet<Test>>::put(vec![validator.clone()]);
+
+		ClusterToValidate::<Test>::put(cluster_id);
```

Comment:
to check that OCW "worked" we shouldn't call create_billing_reports but check if something happened as a result of the run

Reply:
We are calling `create_billing_reports` because we need `active_billing_reports` in `get_era_to_validate`, which is  being called in OCW.